bpython/curtsiesfrontend/repl.py

```diff
@@ -1154,7 +1154,7 @@ def toggle_file_watch(self):
     def add_normal_character(self, char, narrow_search=True):
         if len(char) > 1 or is_nop(char):
             return
-        if self.incr_search_mode and narrow_search:
```

Comment:
@thomasballinger okay. this was the culprit. last commit should fix the option-r behavior. it inserted the closing bracket everytime because the bad condition forced it to never go inside the add_to_incremental_search method when autocompleting brackets. facepalm . i removed the bad condition part and it seems to work just fine now.

Reply:
@thomasballinger did you look at it?